### DISCUSSION TITLE
Elsa 3.1 - Is Posible to Create Custom Class as Variable in designer

### DISCUSSION THREAD

**Discussion by nurhidayat-agung** [elsa-workflows/elsa-core#5557] [Q&A]
hello elsa,
is posible to create custom class variable and import it in elsa studio designer ?
i want create custom class and import it in elsa designer and make to parse from http end point json post body with C# expression

thx.

**Comment by glime-ai** [comment]
Yes, it is possible to create a custom class variable and import it into the Elsa Studio designer. To achieve this, you need to define your custom class in your project and then register it with Elsa so that it becomes available in the designer.

Here's a general outline of the steps you would follow:

1. Define your custom class in your project. For example:
   ```csharp
   public class MyCustomClass
   {
       public string Property1 { get; set; }
       public int Property2 { get; set; }
       // Add other properties as needed
   }
   ```

2. Register your custom class with Elsa's workflow management service so that it becomes available as a variable type in the designer. You can do this in your `Program.cs` or `Startup.cs` file, depending on your project setup:
   ```csharp
   services.AddElsa(elsa => 
   {
       elsa.UseWorkflowManagement(management => 
       {
           management.AddVariableType<MyCustomClass>(category: "Custom");
       });
   });
   ```

3. In the Elsa designer, you can now create a variable of your custom type and use it within your workflows. When you receive a JSON payload from an HTTP endpoint, you can parse it and assign the values to an instance of your custom class using C# expressions.

For more detailed information and examples, you can refer to the Elsa Documentation Website, specifically the sections on [custom types](https://v3.elsaworkflows.io/docs/core-concepts/variables#custom-types) and [variables](https://v3.elsaworkflows.io/docs/core-concepts/variables).

Please note that the exact code and steps may vary depending on the version of Elsa you are using and the specifics of your project setup. If you encounter any issues or need further assistance, feel free to ask for clarification.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by nurhidayat-agung** [reply]
then how to parse json body from http end point to variable using c# expression ?

**Comment by nurhidayat-agung** [comment]
then how to parse json body from http end point to variable using c# expression ?
